apps/skilavottord/ws/src/app/modules/recyclingPartner/recyclingPartner.resolver.ts

```diff
@@ -20,62 +28,38 @@ export class RecyclingPartnerResolver {
     RecyclingPartnerModel[]
   > {
-    return await this.recyclingPartnerService.findActive()
+    return this.recyclingPartnerService.findAllActive()
   }
 
-  @Authorize({ roles: [Role.developer, Role.recyclingFund] })
-  @Mutation((_) => Boolean)
+  @Query(() => RecyclingPartnerModel)
+  async skilavottordRecyclingPartner(
+    @Args('input', { type: () => RecyclingPartnerInput })
+    { companyId }: RecyclingPartnerInput,
+  ): Promise<RecyclingPartnerModel> {
+    return this.recyclingPartnerService.findOne(companyId)
+  }
+
+  @Mutation(() => RecyclingPartnerModel)
   async createSkilavottordRecyclingPartner(
-    @Args('companyId') nationalId: string,
-    @Args('companyName') companyName: string,
-    @Args('address') address: string,
-    @Args('postnumber') postnumber: string,
-    @Args('city') city: string,
-    @Args('website') website: string,
-    @Args('phone') phone: string,
-    @Args('active') active: boolean,
+    @Args('input', { type: () => CreateRecyclingPartnerInput })
+    input: CreateRecyclingPartnerInput,
   ) {
-    const rp = new RecyclingPartnerModel()
-    rp.companyId = nationalId
-    rp.companyName = companyName
-    rp.address = address
-    rp.postnumber = postnumber
-    rp.city = city
-    rp.website = website
-    rp.phone = phone
-    rp.active = active
-    logger.info('create new recyclingPartner...' + JSON.stringify(rp, null, 2))
-    await this.recyclingPartnerService.createRecyclingPartner(rp)
-    return true
+    return this.recyclingPartnerService.createRecyclingPartner(input)
   }
 
-  @Authorize({ roles: [Role.developer, Role.recyclingFund] })
-  @Mutation((_) => String)
-  async skilavottordDeactivateRecycllingPartner(
-    @Args('companyId') nationalId: string,
+  @Mutation(() => RecyclingPartnerModel)
+  async updateSkilavottordRecyclingPartner(
+    @Args('input', { type: () => UpdateRecyclingPartnerInput })
+    input: UpdateRecyclingPartnerInput,
   ) {
-    logger.info('deactivate recyclingPartner:' + nationalId)
-    RecyclingPartnerModel.findOne({ where: { companyId: nationalId } }).then(
-      (rp) => {
-        rp.active = false
-        return rp.save()
-      },
-    )
-    return nationalId
+    return this.recyclingPartnerService.updateRecyclingPartner(input)
   }
 
-  @Authorize({ roles: [Role.developer, Role.recyclingFund] })
-  @Mutation((_) => String)
-  async skilavottordActivateRecycllingPartner(
-    @Args('companyId') nationalId: string,
+  @Mutation(() => Boolean)
+  async deleteSkilavottordRecyclingPartner(
+    @Args('input', { type: () => DeleteRecyclingPartnerInput })
+    input: DeleteRecyclingPartnerInput,
   ) {
-    logger.info('activate recyclingPartner:' + nationalId)
-    RecyclingPartnerModel.findOne({ where: { companyId: nationalId } }).then(
-      (rp) => {
-        rp.active = true
-        return rp.save()
-      },
-    )
-    return nationalId
+    return this.recyclingPartnerService.deleteRecyclingPartner(input)
```

Comment:
Let's call `this.recyclingPartnerService.findOne` first in order to check if it exists, for better error message.

Reply:
Commenting this because the end user (recyclingFund) will already be working with fixed known ids that we use to identify our database records... this might get error prone in terms of UX.


But maybe the error message that `(delete|update|create)RecyclingPartner` returns are clear enough already ... worth checking it out. So in that case, you can ignore my comments.